frame/support/procedural/src/pallet/expand/constants.rs

```diff
@@ -88,35 +85,19 @@ pub fn expand_constants(def: &mut Def) -> proc_macro2::TokenStream {
 				#frame_support::sp_std::marker::PhantomData<(#type_use_gen)>
 			);
 
-			impl<#type_impl_gen> #frame_support::dispatch::DefaultByte for
-				#default_byte_getter<#type_use_gen>
+			impl<#type_impl_gen> #default_byte_getter<#type_use_gen>
 				#completed_where_clause
 			{
 				fn default_byte(&self) -> #frame_support::sp_std::vec::Vec<u8> {
 					#default_byte_impl
 				}
 			}
 
-			unsafe impl<#type_impl_gen> Send for #default_byte_getter<#type_use_gen>
-				#completed_where_clause
-			{}
-			unsafe impl<#type_impl_gen> Sync for #default_byte_getter<#type_use_gen>
-				#completed_where_clause
-			{}
-
-			#frame_support::dispatch::ModuleConstantMetadata {
-				name: #frame_support::dispatch::DecodeDifferent::Encode(#ident_str),
-				ty: #frame_support::dispatch::DecodeDifferent::Encode(#const_type_str),
-				value: #frame_support::dispatch::DecodeDifferent::Encode(
-					#frame_support::dispatch::DefaultByteGetter(
-						&#default_byte_getter::<#type_use_gen>(
-							#frame_support::sp_std::marker::PhantomData
-						)
-					)
-				),
-				documentation: #frame_support::dispatch::DecodeDifferent::Encode(
-					&[ #( #doc ),* ]
-				),
+			#frame_support::metadata::PalletConstantMetadata {
+				name: #ident_str,
+				ty: #frame_support::scale_info::meta_type::<#const_type>(),
+				value: #default_byte_getter::<#type_use_gen>(Default::default()).default_byte(),
```

Comment:
we can even get rid of default_byte_getter and do this:
Anyway we can do in follow-up
```suggestion
				value: { #default_byte_impl },
```

Reply:
Done